### DISCUSSION TITLE
Unable to get the RIGHT timestamp, what can I do?

### DISCUSSION THREAD

**Discussion by BraveSpeaker** [openai/whisper#1819] [Q&A]
Hello,
I am trying to transcribe the following video: ....
At 1:37 the speaker says: " A company built hundreds of castles.. ".
However, whichever the model or method I am using, I am never able to get that timestimp right (except for tiny.en which has a very bad transcription anyway),
Here are the experiments I did:

Used the file.mp4 then mp3 to see if there is a difference, models used:
- Model used -> Timestamp saved for the original 1:37 speech (A company built hundreds of castles..)
- medium.es -> 1:27
- large-v3 -> 1:30
- large -> 1:30
- Small.en -> 1.27
- base.en -> 1:22
- tiny.en -> 1:37 ( THIS IS THE ONLY CORRECT ONE, but the transcription is bad)
I tried on just the audio mp3 to see if there was any difference:
- medium.en -> 1:27
Tried to separate vocals from musics:
- medium.en -> 1:27
What can I do to ensure getting the right timestamps while keeping correct transcriptions of higher models?

**Comment by glangford** [comment]
Try `--model medium.en --word_timestamps True --output_format srt` which generates following (see 1:36.5).

1
00:00:00,000 --> 00:00:04,300
Hello everyone this is your Daily Dose of Internet. Someone made this incredibly

2
00:00:04,300 --> 00:00:05,820
realistic wolf mask.

3
00:00:22,940 --> 00:00:27,040
A car was out of control and rolling down a hill when someone jumped through

4
00:00:27,040 --> 00:00:28,200
the window to stop it.

5
00:00:38,120 --> 00:00:42,560
I found this video of a really bouncy minivan.

6
00:00:58,520 --> 00:01:03,160
One of my subscribers sent me a video of a spider bouncing on their web. Some

7
00:01:03,160 --> 00:01:06,380
spiders will bounce like this as a defense mechanism when they are scared.

8
00:01:17,360 --> 00:01:21,380
These food delivery robots seem to keep having issues.

9
00:01:36,500 --> 00:01:40,860
A company built hundreds of mini castles to turn them into luxury homes

10
00:01:40,860 --> 00:01:45,000
but the company went bankrupt in 2018 and they still stand there today totally

11
00:01:45,000 --> 00:01:49,260
abandoned. And we're like in the mountains too so it's like beautiful out

12
00:01:49,260 --> 00:01:58,440
here. This is pretty scary how would you like to run into this guy underwater?

13
00:02:06,160 --> 00:02:10,100
We now have the ability to turn humans into diamonds by using the cremated

14
00:02:10,100 --> 00:02:14,280
ashes from a loved one. The ashes then go into a custom-made graphite crucible

15
00:02:14,280 --> 00:02:18,860
where it is turned into pure carbon. It's high pressure high temperature and over

16
00:02:18,860 --> 00:02:23,480
time it's gonna be transformed into a raw diamond. One of our master cutters

17
00:02:23,480 --> 00:02:27,980
takes the rough diamond and cuts the exact shape and size out of that rough

18
00:02:27,980 --> 00:02:32,100
diamond. My only goal in life is to be as happy and carefree as this little bird.

19
00:02:32,600 --> 00:02:35,520
That's the end of this video I hope everyone has an amazing rest of their

20
00:02:35,520 --> 00:02:38,620
day and I'll see you guys again very very soon. Later!

**Comment by BraveSpeaker** [comment]
Ah thank you very much, the output did not affect the results right? (because the srt is generated anyway)
It's this part (--word_timestamps True)
I thought this one was about have timestamps for EVERY word, no? Isn't there an option for that? I think I am missing something here

**Reply by glangford** [reply]
> I thought this one was about have timestamps for EVERY word

This does generate timestamps for every word and that process makes the .srt timings more accurate.

**Reply by BraveSpeaker** [reply]
So there is no feature that gives you an SRT containing only "ONE word" per time stamp?

**Reply by glangford** [reply]
Try `--max_words_per_line`

**Reply by BraveSpeaker** [reply]
--max_words_per_line 1
gave full sentences?

**Reply by glangford** [reply]
are you looking at the console output or the .srt

* https://github.com/openai/whisper/discussions/1808

**Reply by glangford** [reply]
p.s. see `whisper --help`
